### DISCUSSION TITLE
Какой стек технологий мы будем использовать?

### DISCUSSION THREAD

**Discussion by temikfart** [CoolStory-Studio/coolstory-gram#1] [Техническое]
Нужно решить какой стек технологий мы будем использовать для написания нашего Мессенджера. Итак, нам нужно:

### База:
- **Язык:** С++ (стандарт??)
- **Система сборки:** [CMake](https://cmake.org/)
- **Протокол передачи данных:** [TCP/IP](https://en.wikipedia.org/wiki/Internet_protocol_suite) + [WebSocket](https://en.wikipedia.org/wiki/WebSocket)
- **Протокол шифрования данных:** [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security)
- **Контейнеризация и деплой:** [Docker](https://www.docker.com/)
- **Unit-тестирование:** [GoogleTest](https://google.github.io/googletest/)

### Дополнительно:
- **БД:** Postgres

### Библиотеки:
- **Поддержка TCP и WebSocket:** [Boost Beast](https://github.com/boostorg/beast)
- **Поддержка шифрования данных:** _???OpenSSL???_
- **Поддержка JSON для упаковки данных:** [nlohmann/json](https://github.com/nlohmann/json#design-goals)
- **Клиент для БД:** [libpqxx](https://github.com/jtv/libpqxx) _(Официальный клиент для PSQL)_
- **Unit-тестирование:** [GoogleTest](https://github.com/google/googletest)

**Comment by cann3v** [comment]
`Boost.Beast` победил... `WebSocket` оказался сильней
